packages/flutter/test/cupertino/date_picker_test.dart

```diff
@@ -2488,29 +2488,70 @@ void main() {
     expect(widths.indexOf(largestWidth), equals(1));
   }, skip: isBrowser); // https://github.com/flutter/flutter/issues/39998
 
-  // Regression test for https://github.com/flutter/flutter/issues/161773
-  testWidgets('CupertinoDatePicker date value baseline alignment', (WidgetTester tester) async {
-    await tester.pumpWidget(
-      CupertinoApp(
-        home: Center(
-          child: SizedBox(
-            width: 400,
-            height: 400,
-            child: CupertinoDatePicker(
-              mode: CupertinoDatePickerMode.date,
-              onDateTimeChanged: (_) {},
-              initialDateTime: DateTime(2025, 2, 14),
-            ),
-          ),
+  testWidgets('showTimeSeperator is only supported in time or dateAndTime mode', (
+    WidgetTester tester,
+  ) async {
+    expect(
+      () => CupertinoDatePicker(
+        mode: CupertinoDatePickerMode.time,
+        onDateTimeChanged: (DateTime _) {},
+        showTimeSeperator: true,
+      ),
+      returnsNormally,
+    );
+
+    expect(
+      () => CupertinoDatePicker(
+        mode: CupertinoDatePickerMode.dateAndTime,
+        onDateTimeChanged: (DateTime _) {},
+        showTimeSeperator: true,
+      ),
+      returnsNormally,
+    );
+
+    expect(
+      () => CupertinoDatePicker(
+        mode: CupertinoDatePickerMode.date,
+        onDateTimeChanged: (DateTime _) {},
+        showTimeSeperator: true,
+      ),
+      throwsA(
+        isA<AssertionError>().having(
+          (AssertionError e) => e.message ?? 'Unknown error',
+          'message',
+          contains('showTimeSeperator is only supported in time or dateAndTime modes'),
         ),
       ),
     );
 
-    Offset lastOffset = tester.getTopLeft(find.text('November'));
-    expect(tester.getTopLeft(find.text('11')).dy, lastOffset.dy);
+    expect(
+      () => CupertinoDatePicker(onDateTimeChanged: (DateTime _) {}),
+      returnsNormally,
+    );
+    
+    // Regression test for https://github.com/flutter/flutter/issues/161773
+    testWidgets('CupertinoDatePicker date value baseline alignment', (WidgetTester tester) async {
```

Comment:
Looks like this test is inside of the new `'showTimeSeperator is only supported in time or dateAndTime mode'` test. This should be moved outside.